packages/cli/yaml/yaml-schema/src/utils/visitRawTypeReference.ts

```diff
@@ -20,28 +23,79 @@ export interface RawTypeReferenceVisitor<R> {
     unknown: () => R;
 }
 
-export function visitRawTypeReference<R>(type: string, visitor: RawTypeReferenceVisitor<R>): R {
+export function visitRawTypeReference<R>({
+    type,
+    _default,
+    validation,
+    visitor
+}: {
+    type: string;
+    _default?: unknown;
+    validation?: ValidationSchema;
+    visitor: RawTypeReferenceVisitor<R>;
+}): R {
     switch (type) {
-        case RawPrimitiveType.integer:
+        case RawPrimitiveType.integer: {
+            const maybeNumberValidation = validation != null ? (validation as NumberValidationSchema) : undefined;
             return visitor.primitive({
                 v1: PrimitiveTypeV1.Integer,
-                v2: undefined // TODO: Add rules for integer.
+                v2: PrimitiveTypeV2.integer({
+                    default: _default != null ? (_default as number) : undefined,
+                    validation:
+                        maybeNumberValidation != null
+                            ? {
+                                  min: maybeNumberValidation.min,
+                                  max: maybeNumberValidation.max,
+                                  exclusiveMin: maybeNumberValidation.exclusiveMin,
+                                  exclusiveMax: maybeNumberValidation.exclusiveMax,
+                                  multipleOf: maybeNumberValidation.multipleOf
+                              }
+                            : undefined
+                })
             });
-        case RawPrimitiveType.double:
+        }
+        case RawPrimitiveType.double: {
+            const maybeNumberValidation = validation != null ? (validation as NumberValidationSchema) : undefined;
             return visitor.primitive({
                 v1: PrimitiveTypeV1.Double,
-                v2: undefined // TODO: Add rules for double.
+                v2: PrimitiveTypeV2.double({
+                    default: _default != null ? (_default as number) : undefined,
+                    validation:
+                        maybeNumberValidation != null
+                            ? {
+                                  min: maybeNumberValidation.min,
+                                  max: maybeNumberValidation.max,
+                                  exclusiveMin: maybeNumberValidation.exclusiveMin,
+                                  exclusiveMax: maybeNumberValidation.exclusiveMax,
+                                  multipleOf: maybeNumberValidation.multipleOf
+                              }
+                            : undefined
+                })
+            });
+        }
+        case RawPrimitiveType.string: {
+            const maybeStringValidation = validation != null ? (validation as StringValidationSchema) : undefined;
+            return visitor.primitive({
+                v1: PrimitiveTypeV1.String,
+                v2: PrimitiveTypeV2.string({
+                    default: _default != null ? (_default as string) : undefined,
```

Comment:
do we have a `fern check` rule that checks the default values against the `type` ? otherwise this cast can thread through incorrect types

Reply:
I was originally planning to add it in a follow-up, but I went ahead and included it here